tests/test_all_rules.py

```diff
@@ -455,6 +456,30 @@ def test_required_lookback(self):
             err_msg = f'The following rules should have a longer `from` defined, due to indexes used\n {rules_str}'
             self.fail(err_msg)
 
+    def test_eql_lookback(self):
+        """Ensure EQL rules lookback => max_span, when defined."""
+        unknowns = []
+        invalids = []
+
+        for rule in self.all_rules:
+            if rule.contents.data.type == 'eql' and rule.contents.data.max_span:
+                if rule.contents.data.look_back == 'unknown':
+                    unknowns.append(self.rule_str(rule, trailer=None))
+                else:
+                    look_back = rule.contents.data.look_back.as_milliseconds()
+                    max_span = rule.contents.data.max_span.as_milliseconds()
+
+                    if look_back < max_span:
+                        invalids.append(f'{self.rule_str(rule)} lookback: {look_back}, maxspan: {max_span}')
```

Comment:
I'm wondering if we can/should do this approach here and in other places. Then we don't have to collect the failures, batch them up and do a custom self.fail later. Thoughts? 
https://stackoverflow.com/a/60309626

Reply:
similar to my comment [here](https://github.com/elastic/detection-rules/pull/1361/files#r673393693), for rules, stack trace is very rarely relevant to failures. I don't know that there is any benefit to breaking them completely out into individual failure stacks vs aggregating like this. Do you have any specific examples where this would be necessary?